engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -125,31 +126,83 @@ object WorkflowActor {
 
   case object Restart extends WorkflowActorMessage with StartMode {
```

Comment:
It's only here because that was the obvious place for it given our current design.  I'm happy to hear suggestions for alternatives or I can try to think of some myself... though considering there's this whole shadow world emerging, I wonder if it makes sense to figure this out now or figure it out when we move this to the shadow world